xbmc/input/InputCodingTable.h

```diff
@@ -30,6 +30,26 @@ class IInputCodingTable
   virtual int GetType() { return TYPE_WORD_LIST; }
 
   virtual ~IInputCodingTable() {}
+  /*! \brief Called for the active keyboard layout when it's loaded, stick any initialization here
+
+      This won't be needed for most implementations so we don't set it =0 but provide a default
+      implementation.
+
+      \return true on success, false on any fatal errors
+  */
+  virtual bool Initialize() { return true; }
+
+  /*! \brief Called for the active keyboard layout when it's unloaded, stick any cleanup here
+
+      This won't be needed for most implementations so we don't set it =0 but provide a default
+      implementation.
+  */
+  virtual void UnInitialize() {}
+
+  /*! \brief Can be overridden if initialization is expensive to avoid calling initialize more than needed
+      \return true if initialization has beeen done and was successful, false otherwise.
+  */
+  virtual bool IsInitialized() const { return false; }
```

Comment:
There isn't really an owning class for it, it stores the current input table but we can have several initialized so it makes more sense to store the state in the input table.
Currently we only have one implementation that needs it, the others can be left untouched as I've provided default implementations of them